server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DatabasePermissionController.java

```diff
@@ -0,0 +1,103 @@
+import com.oceanbase.odc.service.common.response.ListResponse;
+import com.oceanbase.odc.service.common.response.PaginatedResponse;
+import com.oceanbase.odc.service.common.response.Responses;
+import com.oceanbase.odc.service.common.response.SuccessResponse;
+import com.oceanbase.odc.service.permission.database.DatabasePermissionService;
+import com.oceanbase.odc.service.permission.database.model.CreateDatabasePermissionReq;
+import com.oceanbase.odc.service.permission.database.model.DatabasePermissionModel;
+import com.oceanbase.odc.service.permission.database.model.DatabasePermissionType;
+import com.oceanbase.odc.service.permission.database.model.QueryDatabasePermissionParams;
+
+import io.swagger.annotations.ApiOperation;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/1/4 10:15
+ */
+@RestController
+@RequestMapping("api/v2/collaboration/projects/{projectId:[\\d]+}/databasePermissions")
+public class DatabasePermissionController {
+
+    @Autowired
+    private DatabasePermissionService service;
+
+    @ApiOperation(value = "listDatabasePermissions", notes = "List database permissions")
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    public PaginatedResponse<DatabasePermissionModel> list(@PathVariable Long projectId,
+            @RequestParam(name = "user") Long userId,
+            @RequestParam(name = "database", required = false) String fuzzyDatabaseName,
+            @RequestParam(name = "datasource", required = false) String fuzzyDatasourceName,
+            @RequestParam(name = "environment", required = false) List<Long> environmentIds,
+            @RequestParam(name = "permissionType", required = false) List<DatabasePermissionType> permissionTypes,
+            @RequestParam(name = "sourceType", required = false) List<PermissionSourceType> sourceTypes,
+            @RequestParam(name = "expired", required = false) List<Boolean> expiredList,
+            @RequestParam(name = "expireSoon", required = false) Boolean expireSoon,
+            @PageableDefault(size = Integer.MAX_VALUE, sort = {"id"}, direction = Direction.DESC) Pageable pageable) {
+        Boolean expired = CollectionUtils.size(expiredList) == 1 ? expiredList.get(0) : null;
+        QueryDatabasePermissionParams params = QueryDatabasePermissionParams.builder()
+                .userId(userId)
+                .fuzzyDatabaseName(fuzzyDatabaseName)
+                .fuzzyDatasourceName(fuzzyDatasourceName)
+                .environmentIds(environmentIds)
+                .permissionTypes(permissionTypes)
+                .sourceTypes(sourceTypes)
+                .expired(expired)
+                .expiredSoon(expireSoon)
+                .build();
+        return Responses.paginated(service.list(projectId, params, pageable));
+    }
+
+    @ApiOperation(value = "createDatabasePermissions", notes = "Create database permissions")
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
```

Comment:
`/api/v2/collaboration/projects/{projectId:[\\d]+}/databasePermissions` + `POST` is enough? `/create` seems redundant.